### PR TITLE
Setup Psalm and reuse upstream workflows

### PR DESCRIPTION
**Pull request by greg0ire**

### CODE REVIEW

File: psalm.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0"?>
+<psalm
+    errorLevel="7"
```
Comment (stof) [inline psalm.xml:3]: isn't it quite high as level (knowing that for psalm, higher levels are less strict) ?

Comment (derrabus) [reply psalm.xml:3]: I assume, @greg0ire wants to start easy, so we can iterate further once the CI job is in place.

**Review by derrabus — APPROVED** [review]

File: psalm.xml (hunk 1 above)
Comment (greg0ire) [reply psalm.xml:3]: I let Psalm pick the level itself with `psalm --init`. Level 6 finds 40 errors, which would make this PR a bit fat IMO.

Comment (greg0ire) [reply psalm.xml:3]: Yes exactly @derrabus . Also, this allows me to reuse all upstream workflows: the static analysis workflow cannot be reused if we don't have Psalm in the project.

**Review by SenseException — APPROVED** [review]

**Pull request merged**